### PR TITLE
Add one-hot target test coverage for Trainer.evaluate accuracy

### PR DESCRIPTION
**Pull request by Shashank-Tripathi-07**
## What

`Trainer.evaluate`'s accuracy computation branches on whether targets are integer class indices (1D) or one-hot encoded (2D), with its own comment documenting both as supported:

```python
if len(targets.data.shape) == 1:  # Integer targets
    correct += np.sum(predictions == targets.data)
else:  # One-hot targets
    correct += np.sum(predictions == np.argmax(targets.data, axis=1))
```

Every existing classification test in this suite (`tests/08_training/test_training_coverage.py`) used integer targets only, leaving the one-hot branch completely untested.

## Why

Found while triaging mutation-testing survivors for `08_training`. That line had 9 unkillable survivor mutants.

## A note on scope

`CrossEntropyLoss.forward` only accepts integer class-index targets per its own docstring (`targets.data.astype(int)` on a one-hot 2D array produces nonsense), so it can't actually be paired with one-hot targets in practice. The new tests use a minimal stub loss to isolate `evaluate()`'s own accuracy logic from that unrelated limitation, since `evaluate()`'s one-hot handling exists specifically for the accuracy computation and is independent of which loss function is in use.

## Fix

Added:
- `test_classification_accuracy_with_onehot_targets` / `test_zero_accuracy_for_wrong_predictions_onehot_targets`: one-hot counterparts of the existing integer-target perfect/wrong-classifier tests
- `test_mixed_onehot_batch_partial_accuracy`: a batch with some correct and some incorrect one-hot predictions, exercising the `np.sum(...)` count itself (fractional accuracy) rather than only the two all-correct/all-wrong extremes

## Verification (hand-mutation)

The one-hot argmax's `axis=1` flipped to `axis=0`: previously had zero effect on any test. Now caught (2 test failures, both surfacing the resulting shape-mismatch `ValueError`).

Full `tests/08_training` suite (62 tests) still passes.

### CODE REVIEW

**Comment by Shashank-Tripathi-07** [comment]
Closing: this PR adds test coverage for already-correct behavior (found via mutation testing/fuzzing, no actual bug), not a fix. Reversing course on pure coverage-expansion PRs per maintainer guidance to keep the surface area of open PRs limited to real fixes.
